src/layouts/Settings/PrivacySettings.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  VStack,
+  FormControl,
+  Box,
+  Input,
+  Flex,
+  InputGroup,
+  Icon,
+  InputRightElement,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverArrow,
+  PopoverBody,
+  Text,
+} from "@chakra-ui/react"
+import {
+  FormErrorMessage,
+  FormLabel,
+  IconButton,
+} from "@opengovsg/design-system-react"
+import { useEffect, useState } from "react"
+import { useFormContext, useFormState } from "react-hook-form"
+import { BiCopy } from "react-icons/bi"
+
+import { Section, SectionHeader } from "layouts/components"
+
+import { generatePassword } from "utils/password"
+import { PASSWORD_REGEX } from "utils/validators"
+
+import { SitePasswordSettings } from "types/settings"
+
+import { FormToggle } from "./components/FormToggle"
+
+interface PrivacySettingsProp {
+  isError: boolean
+}
+
+export const PrivacySettings = ({
+  isError,
+}: PrivacySettingsProp): JSX.Element => {
+  const { register, trigger, watch, setValue, getValues } = useFormContext()
+  const { errors } = useFormState<SitePasswordSettings>()
+  const [hasCopied, setHasCopied] = useState(false)
+  const [isBuffering, setIsBuffering] = useState(false)
+
+  const generateAcceptablePassword = () => {
+    const regex = new RegExp(PASSWORD_REGEX)
+    let password = ""
+    while (!regex.test(password)) {
+      password = generatePassword()
+    }
+    return password
+  }
+
+  const isPrivatised = watch("privatiseStaging")
+
+  useEffect(() => {
+    // Generate a default password on toggle if no password exists
+    if (!getValues("password")) {
+      const newPassword = generateAcceptablePassword()
+      setValue("password", newPassword)
+      trigger("password")
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isPrivatised])
+
+  return (
+    <Section id="privacy-fields">
+      <SectionHeader label="Privacy Settings" />
+      <VStack spacing="1.5rem" align="flex-start" w="50%">
+        <FormControl isDisabled={isError}>
+          <Flex justifyContent="space-between" w="100%">
+            <FormLabel isRequired>Privatise staging site</FormLabel>
+            {/* NOTE: This should be toggle from design system
+                but the component is broken and doesn't display a slider */}
```

Comment:
Shall we add a todo to replace this after we up design system?

Reply:
I've added the TODO, but for a slightly different change - after having a chat with Nat, we're opting to swap all the existing toggle buttons in settings to checkboxes instead! Have added a TODO for that instead to swap all at one time since that is out of scope of this PR